File: GoogleInterview.cs
using System;
using System.Collections.Generic;

public class Program
{
    public static void Main()
    {
        string[] array1 = { "a", "b", "c", "x" };
        string[] array2 = { "z", "y", "a" };

        Console.WriteLine(ContainsCommonItem(array1, array2));
    }

    public static bool ContainsCommonItem(string[] arr1, string[] arr2)
    {
        Dictionary<string, bool> map = new Dictionary<string, bool>();

        foreach (string item in arr1)
        {
            if (!map.ContainsKey(item))
            {
                map[item] = true;
            }
        }

        foreach (string item in arr2)
        {
            if (map.ContainsKey(item))
            {
                return true;
            }
        }

        return false;
    }
}

File: GoogleInterview.go
package main

import (
	"fmt"
)

func main() {
	array1 := []string{"a", "b", "c", "x"}
	array2 := []string{"z", "y", "a"}

	fmt.Println(containsCommonItem(array1, array2))
}

func containsCommonItem(arr1 []string, arr2 []string) bool {
	m := make(map[string]bool)

	for _, item := range arr1 {
		if _, ok := m[item]; !ok {
			m[item] = true
		}
	}

	for _, item := range arr2 {
		if _, ok := m[item]; ok {
			return true
		}
	}

	return false
}
